lib/datadog/core/configuration/components.rb

```diff
@@ -81,13 +82,17 @@ def build_telemetry(settings, agent_settings, logger)
         def initialize(settings)
           @logger = self.class.build_logger(settings)
 
+          # This agent_settings is intended for use within Core. If you require
+          # agent_settings within a product outside of core you should extend
+          # the Core resolver from within your product/component's namespace.
           agent_settings = AgentSettingsResolver.call(settings, logger: @logger)
 
           @remote = Remote::Component.build(settings, agent_settings)
-          @tracer = self.class.build_tracer(settings, agent_settings)
+          @tracer = self.class.build_tracer(settings, logger: @logger)
+
           @profiler = Datadog::Profiling::Component.build_profiler_component(
             settings: settings,
-            agent_settings: agent_settings,
+            logger: @logger,
             optional_tracer: @tracer,
           )
```

Comment:
I'd like to request that the profiler still receives the `agent_settings` from the core, as I don't plan on maintaining any profiling-specific changes.

(My whole intention with the creation of the `AgentSettingsResolver` was exactly to have one solution that fits most/all products, rather than needing to keep a separate resolver)

Reply:
What do you think about profiling using the core resolver, but having the profiler component instantiating it, rather than having it passed in? 

`build_profiler_component` would change from

`agent_settings = Datadog::Profiling::AgentSettingsResolver.call(settings, logger: logger)`
to
`agent_settings = Datadog::Core::Configuration::AgentSettingsResolver.call(settings, logger: logger)`.

I believe we are going to look at ways to decouple [Core::Configuration::Components](https://github.com/DataDog/dd-trace-rb/blob/master/lib/datadog/core/configuration/components.rb) prior to 2.0. I think part of that will entail finding a way for components to better encapsulate their own initialization.

Reply:
Passing the existing `AgentSettings` to profiling for now doesn't seem problematic, especially since profiling isn't seeking any custom behavior.  I can create a follow-up PR that will revert `build_profiler_component` to its previous state. 

I think we're going to have to rethink how we instantiate components and how they get settings sooner, rather than later. But, we can cross that bridge when we get there.